### ISSUE TITLE
Python 3.8+ issue with #undef _POSIX_THREADS

### ISSUE THREAD

**Issue by he-hesce** [MeVisLab/pythonqt#105]
Compiling code using PythonQt release 3.4.1 on Rocky Linux 9.2 with Python 3.9 gives the following compile error:

```
/usr/include/python3.9/pythread.h:136:5: error: #error "Require native threads. See https://bugs.python.org/issue31370"
  136 | #   error "Require native threads. See https://bugs.python.org/issue31370"
      |     ^~~~~
/usr/include/python3.9/pythread.h:145:5: error: ‘NATIVE_TSS_KEY_T’ does not name a type
  145 |     NATIVE_TSS_KEY_T _key;
      |     ^~~~~~~~~~~~~~~~
```

By commenting out the `#undef` of `_POSIX_THREADS` (hence keeping `_POSIX_THREADS` defined) in `src/PythonQtPythonInclude.h` it builds and works fine.

```
$ cat /etc/redhat-release 
Rocky Linux release 9.2 (Blue Onyx)

$ python --version
Python 3.9.16

$ qmake --version
QMake version 3.1
Using Qt version 5.15.3 in /usr/lib64

$ gcc --version
gcc (GCC) 11.3.1 20221121 (Red Hat 11.3.1-4)
```

Otherwise release 3.4.1 is working fine (without bespoke fixes/patches) for me now on CentOS 7 (Python 2.7), Rocky 8 (Python 3.6), Rocky 9 (Python 3.9), Ubuntu 20.04 LTS (Python 3.8), and Ubuntu 22.04 LTS (Python 3.10).

Only Python 3.8+ builds are affected by the _POSIX_THREADS issue so that also includes Ubuntu 22.04 LTS (Python 3.10) and Ubuntu 20.04 LTS (Python 3.8) in addition to Rocky 9 (Python 3.9). On all three platforms, not undefining _POSIX_THREADS solves the issue.

**Comment by he-hesce** [comment]
Otherwise release 3.4.1 is working fine (without bespoke fixes/patches) for me now on CentOS 7 (Python 2.7), Rocky 8 (Python 3.6), Rocky 9 (Python 3.9), Ubuntu 20.04 LTS (Python 3.8), and Ubuntu 22.04 LTS (Python 3.10).

Only Python 3.8+ builds are affected by the _POSIX_THREADS issue so that also includes Ubuntu 22.04 LTS (Python 3.10) in addition to Rocky 9 (Python 3.9).

**Comment by he-hesce** [comment]
@mrbean-bremen: Pull request #103 might address this issue?

**Comment by mrbean-bremen** [comment]
Good to know, thanks! I was wondering if we were the only ones with the problem, and had maybe some configuration wrong. Will merge the PR in this case...

**Comment by mrbean-bremen** [comment]
Oops, referenced the wrong number in the commit... Anyway, the PR is merged and 3.4.2 created, closing this.

**Comment by he-hesce** [comment]
@mrbean-bremen: Thanks. I will baseline our production release on 3.4.2 so may it be a long-lived stable one :-) CentOS 7 (Python 2.7), Rocky 8 (Python 3.6), Rocky 9 (Python 3.9), Ubuntu 20.04 LTS (Python 3.8), and Ubuntu 22.04 LTS (Python 3.10). Cheers!
